### ISSUE TITLE
Crash in adapter content module

### ISSUE THREAD

**Issue by s-andrews** [s-andrews/FastQC#1]
We had this trace reported:

```
$ java -version
java version "1.7.0_161"
OpenJDK Runtime Environment (rhel-2.6.12.0.el6_9-x86_64 u161-b00) OpenJDK 64-Bit Server VM (build 24.161-b00, mixed mode) $ /usr/local/fastqc/0.11.5/fastqc ./foo.fastq Started analysis of foo.fastq Analysis complete for foo.fastq $ ls foo.fastq  foo_fastqc.html  foo_fastqc.zip $ rm foo_* $ /usr/local/fastqc/0.11.6/fastqc ./foo.fastq Started analysis of foo.fastq Exception in thread "Thread-1" java.lang.NegativeArraySizeException
         at uk.ac.babraham.FastQC.Modules.AdapterContent$Adapter.expandLengthTo(AdapterContent.java:319)
         at uk.ac.babraham.FastQC.Modules.AdapterContent.processSequence(AdapterContent.java:148)
         at uk.ac.babraham.FastQC.Analysis.AnalysisRunner.run(AnalysisRunner.java:88)
         at java.lang.Thread.run(Thread.java:748)
```

**Comment by s-andrews** [comment]
I had a look at this and it relates to a previous bug we had which modified the adapter module.

The problem is that if we have really short sequences at the start of the file which are shorter than the adapter fragments we're searching for then we can't possibly find a hit.  The last position we could have found a hit is therefore negative.

In the module we expand an array to store the hit positions and we expand this every time we see a sequence which is longer than we'd seen before.  If the first sequence we see is <12bp we'll try to expand to a negative length and we get the crash above.

**Comment by s-andrews** [comment]
I can reproduce this with a sequence file containing just:

```
@test
GGGGG
+
BBBBB
```

..which gives:

```
Exception in thread "Thread-4" java.lang.NegativeArraySizeException
	at uk.ac.babraham.FastQC.Modules.AdapterContent$Adapter.expandLengthTo(AdapterContent.java:319)
	at uk.ac.babraham.FastQC.Modules.AdapterContent.processSequence(AdapterContent.java:148)
	at uk.ac.babraham.FastQC.Analysis.AnalysisRunner.run(AnalysisRunner.java:88)
	at java.lang.Thread.run(Unknown Source)
```

It doesn't happen if we use:

```
@test2
GGGGGGGGGGGGG
+
BBBBBBBBBBBBB
@test
GGGGG
+
BBBBB
```
So it's only triggered if we see a sequence <12bp before we've ever seen one >12bp, which is why none of our tests picked it up.

**Comment by s-andrews** [comment]
I've created the above patch which fixes the immediate issue.  It might be worth coming back to this at some point to go through how we handle adapters of variable length though.  At the moment we only map up to the last possible match for the longest adapter so late hits for short adapters are missed.  This is probably the safest way to handle this, but there may be a better way to do it.  It won't matter for the default setup though as we're using the same lengths of adapters in all cases.

**Comment by s-andrews** [comment]
This is fixed in v0.11.7
